vmaas_sync/vmaas_sync.py

```diff
@@ -103,6 +103,65 @@ def process_cve_list(cves, cves_in_db, impact_id_map):
     return to_insert, to_update, to_delete
 
 
+def process_repo_list(repos, repos_in_db):
+    """Process repo list and evaulate which to delete and insert"""
+    to_insert = repos - repos_in_db
+    to_insert = [(repo,) for repo in to_insert]
+
+    to_delete = repos_in_db - repos
+    to_delete = [(repo,) for repo in to_delete]
+
+    return to_insert, to_delete
+
+
+def insert_repos(cur, repos):
+    """Insert the repos into db"""
+    if repos:
+        LOGGER.info("Inserting %s new repositories", len(repos))
+        execute_values(cur, """insert into repo (name) values %s
+                               on conflict (name) do nothing""", repos, page_size=len(repos))
+
+
+def delete_repos(cur, repos):
+    """Remove unknown repos from db"""
+    if repos:
+        LOGGER.info("Deleting %s repositories", len(repos))
+        ids = execute_values(cur, "select id from repo where name in (%s)", repos, page_size=len(repos), fetch=True)
+        execute_values(cur, "delete from system_repo where repo_id in (%s)", ids, page_size=len(ids))
```

Comment:
I'd do it same way as it's for CVE sync. We should delete automatically only repos that are not used by any system. Repos that are in use by any system should be only reported in logs so we could check them and delete them manually. This prevents from accidentaly deleting all repo information when engine is set to use empty VMaaS instance.